routes/api/contacts.js

```diff
@@ -1,25 +1,115 @@
-})
+const schema = Joi.object({
+  id: Joi.string().required(),
+  name: Joi.string().alphanum().min(5).max(15).required(),
+  email: Joi.string()
+    .email({
+      minDomainSegments: 2,
+      tlds: { allow: ["com", "net", "pl"] },
+    })
+    .required(),
+  phone: Joi.string().pattern(phonePattern).required(),
+});
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/", async (req, res, next) => {
+  try {
+    const contacts = await listContacts();
+    res.json(contacts);
+  } catch (error) {
+    console.log("Error:", error.message);
+  }
+});
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/:contactId", async (req, res, next) => {
+  try {
+    const { contactId } = req.params;
+    const contact = await getContactById(contactId);
+    if (contact) {
+      res.json(contact);
+    } else {
+      res.status(404).json({ message: "Not found" });
+    }
+  } catch (error) {
+    console.log("Error:", error.message);
+  }
+});
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.post("/", async (req, res, next) => {
+  try {
+    const { name, email, phone } = req.body;
 
-module.exports = router
+    const newContactToValidation = {
+      id: uuidv4(),
+      name,
+      email,
+      phone,
+    };
+
+    const newContact = await schema.validateAsync(newContactToValidation);
+
+    addContact(newContact);
+    res
+      .status(201)
+      .json({ message: "Validation Successful", data: newContact });
+  } catch (error) {
+    res.status(400).json({ message: "Please correctly fill all fields" });
+    console.log("Error:", error.message);
+  }
+});
+
+router.delete("/:contactId", async (req, res, next) => {
+  try {
+    const { contactId } = req.params;
+
+    const contacts = await listContacts();
+    // expected output is object with contact data || undefined
+    const contact = contacts.find((contact) => contact.id === contactId);
+    if (contact) {
+      removeContact(contactId);
+      res.json({ message: "Contact deleted" });
+    } else {
+      res.status(404).json({ message: "Not found" });
+    }
+  } catch (error) {
+    console.log("Error:", error.message);
+  }
+});
+
+router.put("/:contactId", async (req, res, next) => {
+  try {
+    const { contactId } = req.params;
+    const { name, email, phone } = req.body;
+
+    const contacts = await listContacts();
+    const contact = contacts.find((contact) => contact.id === contactId);
+
+    if (contact) {
+      const newContactToValidation = { ...contact, name, email, phone };
+      const newContact = await schema.validateAsync(newContactToValidation);
+      updateContact(contactId, newContact);
+      res.json({ message: "Validation Successful", data: newContact });
+    } else {
+      res.status(404).json({ message: "Not found" });
+    }
```

Comment:
```suggestion
     return res.json({ message: "Validation Successful", data: newContact });

      res.status(404).json({ message: "Not found" });
```
Używanie early return statement umożliwi ominięcie bloku else co ułatwi czytanie

Reply:
Spróbuję się przyzwyczaić do early return statement, nie jest mi to obca technika. Zwykle preferowałem if else, a w krótkich przypadkach ternary operator. Tu mam pytanie kiedy early return statement kiedy if else?